cmd/oracled/cmd/register_oracle.go

```diff
@@ -199,3 +197,35 @@ func generateSealedNodeKey(nodePrivKeyPath string) ([]byte, []byte, error) {
 
 	return nodePubKey, nodeKeyRemoteReport, nil
 }
+
+func subscribeApproveOracleRegistrationEvent(conf *config.Config) error {
+	svc, err := oracleservice.New(conf)
```

Comment:
This will create a new service which has including query client, grpc client, etc.
I thought that those resources can be reused in `sendTxRegisterOracle` and `subscribeApproveOracleRegistrationEvent`.
Why don't we have `Reactor` and `Service` for each event?
For example,

```go
type RegisterOracleEventReactor struct {
    GRPCClient() *panacea.GRPCClient
    EnclaveInfo() *sgx.EnclaveInfo
    OracleAcc() *panacea.OracleAccount
    OraclePrivKey() *btcec.PrivateKey
    Config() *config.Config
    QueryClient() panacea.QueryClient
    IPFS() *ipfs.IPFS
    BroadcastTx(txBytes []byte) (int64, string, error)
}

type ApproveOracleRegistrationEventReactor struct {
    Config() *config.Config
    QueryClient() panacea.QueryClient
    EnclaveInfo() *sgx.EnclaveInfo
    OracleAcc() *panacea.OracleAccount
}
```

Then, each event will receive reactor accordingly.

```go
type RegisterOracleEvent struct {
    reactor RegisterOracleEventReactor
}

type ApproveOracleRegistrationEvent struct {
    reactor ApproveOracleRegistrationEventReactor
}
```

How about this?

Reply:
OK. It looks like better.
I will fix soon.

Reply:
Sorry for the confusion.
It's related to our structure of event and service, not cmd itself.
Thank you!

Reply:
I changed like below:
```go
// Service is 'service/service.go'
type Service interface {
	EnclaveInfo() *sgx.EnclaveInfo
	OracleAcc() *panacea.OracleAccount
	OraclePrivKey() *btcec.PrivateKey
	QueryClient() panacea.QueryClient
	Config() *config.Config
	BroadcastTx([]byte) (int64, string, error)
}

// OracleService is 'service/oracle/service.go'
type OracleService interface {
	EnclaveInfo() *sgx.EnclaveInfo
	OracleAcc() *panacea.OracleAccount
	GetAndStoreOraclePrivKey() error
}
```

This one got the name of the interface it uses

Reply:
@H4NLee 
I'd like to discuss this with you.
In my opinion, it is better to declare only one identical interface imported from outside.
Like `BankKeeper`, `AccountKeeper`

So I think it would be better declare service like below:
```
// Service is 'service/service.go'
type Service interface {
	EnclaveInfo() *sgx.EnclaveInfo
	OracleAcc() *panacea.OracleAccount
	OraclePrivKey() *btcec.PrivateKey
	QueryClient() panacea.QueryClient
	Config() *config.Config
	BroadcastTx([]byte) (int64, string, error)
}
```

Reply:
Agree.
The reason why I made interfaces for each event was due to assumption of using different service.
But for now, we don't have to. We can use the `Service` interface defined in `service/service.go`.

Thank you for your efforts :pray:

Reply:
Thank you for review